src/algebraic_geometry/AffineScheme.lean

```diff
@@ -181,6 +185,17 @@ begin
   exact set.image_univ
 end
 
+lemma is_affine_open.image_is_open_immersion {X Y : Scheme} {U : opens X.carrier}
+  (hU : is_affine_open U)
+  (f : X ⟶ Y) [H : is_open_immersion f] : is_affine_open (H.open_functor.obj U) :=
```

Comment:
Hmm, If `H` is a TC argument, I'm afraid this becomes annoying to use, if in the type we have `H.open_functor` occuring.
If `H` were a regular explicit argument, then `is_open_immersion.is_affine_open_open_functor` would also be a descriptive name.

Reply:
The reason is probably more because `H` should not be an explicit argument of `is_open_immersion.open_functor`. We should probably give an alias to morphisms between schemes to enable dot notation and move several definitions away from the `is_open_immersion` namespace.

Reply:
Sounds like a good experiment for a future PR.